akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -255,7 +255,7 @@ final class ActorTestKit private[akka] (
     try {
       Await.result(internalTestKitGuardian.ask { (x: ActorRef[ActorTestKitGuardian.Ack.type]) =>
         ActorTestKitGuardian.StopActor(ref, x)
-      }, max)
+      }(Timeout(max), scheduler), max)
```

Comment:
With the same time it will be a race between the two timeouts but since `AskTimeoutException` is also a `TimeoutException` and will be caught by the same catch below that should be fine.